keps/sig-storage/1845-prioritization-on-volume-capacity/README.md

```diff
@@ -0,0 +1,424 @@
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+With [volume topology-aware
+scheduling](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/volume-topology-scheduling.md) in place,
+the scheduler filters nodes considering volume topology constraints.
+However, scheduler does not try to prioritize nodes in different topologies based on
+volume capacity. This leads to a waste of resources which we should avoid.
+
+This KEP continues the work proposed [here](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/volume-topology-scheduling.md#priority), to enable the scheduler to favor the node which has the most fit PV.
+
+## Motivation
+
+Because we didn't take volume capacity into account when scheduling a pod that
+can run in multiple topologies. A large PV may be used by a PVC with a small
+capacity request, even if there are many suitable small PVs in other
+topologies. PVCs with a large capacity request may not find feasible PVs to use
+if too many large PVs are consumed by PVCs with a small capacity request.
+
+Like CPU, Memory resources, the scheduler should take volume capacity into
+account in scheduling pods to ensure balanced resource usage.
+
+### Goals
+
+- Nodes of topology in which volumes with close capacity are preferred in
+  scheduling
+- Support arbitrary PV topology domain (e.g. hostname, zone, rack)
+
+### Non-Goals
+
+- Prioritizing nodes based on the total available capacity of static provisioned
+  PVs
+- Prioritizing nodes based on the total available number of static provisioned
+  PVs
+- Prioritizing nodes based on the total available capacity for dynamic
+  provisioning PVCs
+
+## Proposal
+
+### User Stories
+
+#### Volumes with close capacity are preferred
+
+As a cluster administrator, I need to prepare some local PVs with different
+storage capacities (e.g. 10Gi, 100Gi, 1Ti) as different workloads may have
+different storage requirements.
+
+A PV with 1Ti capacity on one node has the same chance as a PV on another node
+with 10Gi capacity to be bound with a 10Gi request PVC in scheduling. If nodes
+of topology in which volumes with close capacity are always preferred, we can
+achieve better storage resource usage.
+
+See https://github.com/kubernetes/kubernetes/issues/83323.
+
+#### Different weights for different classes
+
+Given a cluster which has two kinds of local storage classes:
+
+- hdd
+- ssd
+
+Workloads may request two disks, one from `hdd` and one from `ssd`. As
+the ssd storage resource may be relatively scarce, the administrator can
+configure a relatively larger weight for storage class `ssd` and hope it's more
+balanced.
```

Comment:
The weight is the numeric multiplier of the storage class used in the scoring.
Because pods can request multiple PVs from different storage classes. One of them may be more important than the others and can have a higher weight in scoring.